[BLAS::portBLAS backend] Update nrm2 operator call routine

s-Nick opened the pull request:
# Description

The nrm2 operator available in portBLAS requires that result memory is initialized to zero before performing the routine. I updated the routine call accordingly.

# Checklist

## All Submissions

- [x] Do all unit tests pass locally? Attach a log.
[nrm2.txt](https://github.com/oneapi-src/oneMKL/files/13002778/nrm2.txt)


- [x] Have you formatted the code using clang-format?

s-Nick committed:
[portblas] Update nrm2 operator call routine

Portblas implementation for nrm2 operator requires result memory
to be initialized to zero.

muhammad-tanvir-1211 commented on src/blas/backends/portblas/portblas_level1.cxx:
```diff
@@ -299,6 +305,11 @@ sycl::event nrm2(sycl::queue &queue, std::int64_t n, const std::complex<real_t>
+    // before starting the computation.
+    auto init_res_val = queue.submit(
+        [&](sycl::handler &cgh) { cgh.single_task([=]() { result[0] = real_t(0); }); });
+    init_res_val.wait();
```
Instead of waiting on this event here, I think we can add this event to the `dependencies` vector and that should take care of things.

s-Nick committed:
[portblas] Update dependencies vector for USM

Instead of waiting on the initialization event, create a new vector and use it for
routine dependencies.

s-Nick committed:
[portblas] Update asum usm call

Remove wait on the initialization event and create and use a new vector
to handle routine dependencies.

muhammad-tanvir-1211 reviewed (approved):
LGTM, Thanks.

mkrainiuk reviewed (approved):
Looks good to me, thank you!
